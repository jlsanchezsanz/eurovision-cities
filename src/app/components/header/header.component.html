<section>
  <app-menu></app-menu>
</section>

<section class="header__logo">
  <img
    src="../../assets/logo.png"
    alt="Eurovision Services"
    class="logo__img"
  />
</section>
