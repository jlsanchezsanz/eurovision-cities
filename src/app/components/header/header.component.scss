:host {
  display: flex;
  flex-direction: column;
}

.header {
  &__logo {
    display: flex;
    height: 70px;
    justify-content: center;
  }
}

.logo {
  &__img {
    height: 100%;
  }
}
