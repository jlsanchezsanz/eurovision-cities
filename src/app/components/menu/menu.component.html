<button mat-button [matMenuTriggerFor]="menu">Menu</button>
<mat-menu #menu="matMenu">
  <a mat-menu-item routerLink="/">Home</a>
  <a mat-menu-item routerLink="/cities">Cities</a>
</mat-menu>