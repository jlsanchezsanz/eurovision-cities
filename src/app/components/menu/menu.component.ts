import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-menu',
  templateUrl: './menu.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class MenuComponent {}
