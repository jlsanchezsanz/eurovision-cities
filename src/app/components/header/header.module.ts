import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { MenuModule } from '../menu';
import { HeaderComponent } from './header.component';

@NgModule({
  declarations: [HeaderComponent],
  imports: [CommonModule, MenuModule],
  exports: [HeaderComponent]
})
export class HeaderModule {}
