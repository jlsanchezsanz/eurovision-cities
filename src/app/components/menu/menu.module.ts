import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonModule, MatMenuModule } from '@angular/material';
import { RouterModule } from '@angular/router';

import { MenuComponent } from './menu.component';

@NgModule({
  declarations: [MenuComponent],
  imports: [CommonModule, MatButtonModule, MatMenuModule, RouterModule],
  exports: [MenuComponent]
})
export class MenuModule {}
