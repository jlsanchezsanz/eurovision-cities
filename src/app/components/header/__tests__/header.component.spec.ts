import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MockModule } from 'ng-mocks';

import { HeaderComponent } from '../header.component';
import { MenuModule } from '../../menu';

describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HeaderComponent],
      imports: [MockModule(MenuModule)]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
