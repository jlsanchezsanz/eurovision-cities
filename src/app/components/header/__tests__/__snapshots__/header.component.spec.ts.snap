// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match snapshot 1`] = `
<app-header>
  <section>
    <app-menu />
  </section><section
    class="header__logo"
  >
    <img
      alt="Eurovision Services"
      class="logo__img"
      src="../../assets/logo.png"
    />
  </section>
</app-header>
`;
